### ISSUE TITLE
Job recovery when tools have been removed

### ISSUE THREAD

**Issue by natefoo** [galaxyproject/galaxy#5424]
Looks like this broke at some point. Reported by @vipints on Gitter:

```pytb
galaxy.jobs.handler WARNING 2018-01-31 15:36:15,704 (78691) Tool 'motifgen-fasta_1' removed from tool config, unable to recover job
galaxy.jobs DEBUG 2018-01-31 15:36:15,705 (78691) Working directory for job is: /tools/galaxy_ratschlab/database/job_working_directory/078/78691
Traceback (most recent call last):
  File "/tools/galaxy_ratschlab/lib/galaxy/webapps/galaxy/buildapp.py", line 58, in paste_app_factory
    app = galaxy.app.UniverseApplication(global_conf=global_conf, **kwargs)
  File "/tools/galaxy_ratschlab/lib/galaxy/app.py", line 190, in __init__ 
    self.job_manager.start()
  File "/tools/galaxy_ratschlab/lib/galaxy/jobs/manager.py", line 33, in start
    self.job_handler.start()
  File "/tools/galaxy_ratschlab/lib/galaxy/jobs/handler.py", line 40, in start
    self.job_queue.start()
  File "/tools/galaxy_ratschlab/lib/galaxy/jobs/handler.py", line 85, in start
    self.__check_jobs_at_startup()
  File "/tools/galaxy_ratschlab/lib/galaxy/jobs/handler.py", line 140, in __check_jobs_at_startup
    self.job_wrapper(job).fail('This tool was disabled before the job completed.  Please contact your Galaxy administrator.') 
  File "/tools/galaxy_ratschlab/lib/galaxy/jobs/__init__.py", line 993, in fail
    if outputs_to_working_directory and not self.__link_file_check():
  File "/tools/galaxy_ratschlab/lib/galaxy/jobs/__init__.py", line 1747, in __link_file_check
    param_dict = job.get_param_values(self.app)
  File "/tools/galaxy_ratschlab/lib/galaxy/model/__init__.py", line 702, in get_param_values 
    param_dict = tool.params_from_strings(param_dict, app, ignore_errors=ignore_errors)
AttributeError: 'NoneType' object has no attribute 'params_from_strings'`
```

The workaround is to manually error the offending job in the db (note this does not update dataset states in the history), e.g.:

```sql
update job set state='error' where id=78691;
```

**Comment by vipints** [comment]
Thanks @natefoo for creating this ticket! 
As you mentioned, I have manually edit set the job state to `error` for those jobs from our instance. Used the following sql statement on my postgres db. 

`update job set state = 'error' where state = 'new';`

Our Galaxy instance is running with version `release_17.09`. Please let me know if you need any details from my side.

**Comment by mvdbeek** [comment]
So that was sufficient to resume operation ?

**Comment by vipints** [comment]
Yes, It has updated ~ 600 entries from our instance JOB table.

**Comment by mvdbeek** [comment]
OK, we should be able to fix this, thanks for reporting!

**Comment by jmchilton** [comment]
Fixed by #5456 thanks to @mvdbeek
